tools/gfx/gfx.slang

```diff
@@ -1710,6 +1710,19 @@ struct SlangDesc
     slang::SlangLineDirectiveMode lineDirectiveMode = slang::SlangLineDirectiveMode::SLANG_LINE_DIRECTIVE_MODE_DEFAULT;
 };
 
+struct ShaderCacheDesc
+{
+    // The filename for the file the cache's state should be saved to or loaded from.
+    NativeString cacheFilename = "cache.txt";
+    // The root directory for the shader cache.
+    NativeString shaderCachePath = nullptr;
+    // The file system for loading cached shader kernels. The layer does not maintain a strong reference to the object,
+    // instead the user is responsible for holding the object alive during the lifetime of an `IDevice`.
+    void* shaderCacheFileSystem = nullptr;
+    // The maximum number of entries stored in the cache.
+    GfxCount entryCountLimit = 1000;
```

Comment:
Now that I think about it, a 0 default here might be more appropriate, since caching is off by default.

Reply:
Okay, I suppose this (and the comment above) also makes sense from a consistency perspective. Are we expecting the user to provide both a filename and entry count limit then if they're choosing to turn caching on?

Reply:
A 0 limit should mean no limit, which is also valid. The user should specify a limit if they wish to limit.

Reply:
Changed this to 0, but I'm opting to keep the default on the cache filename since the cache is enabled by providing at least one of `shaderCachePath` and `shaderCacheFileSystem`. If someone wants to enable the cache but doesn't care about limiting entry count or exactly what the filename is, then these defaults allow them to do that.

Will be in next push